### PR TITLE
Switch from vee-validate to Vuelidate

### LINKED ISSUE #334: Switch from vee-validate to vuelidate
### Problem

- vee-validate injects `errors` into every component (although this can apparently be disabled via the config by setting `inject` to false)
- vee-validate has kinda an ugly API
- vee-validate is huge, the main .min file is [72.2kb](https://github.com/baianat/vee-validate/blob/master/dist/vee-validate.min.js), and the minimal version is still a whopping [33.3kb](https://github.com/baianat/vee-validate/blob/master/dist/vee-validate.minimal.min.js).

### Solution

Look into [other validation plugins](https://github.com/vuejs/awesome-vue#validation).

Current promising candidate is [vuelidate](https://github.com/monterail/vuelidate).

/cc @hubudibu 

**Update:** Yep, looks like we're switching to vuelidate, so let's replace all of the vee-validate stuff with it.

### PR DESCRIPTION
**Pull request by hubudibu**
Fixes #334 Switch from [vee-validate](http://vee-validate.logaretm.com/) to [vuelidate](https://monterail.github.io/vuelidate/)

affected pages:
- SignUp
- Login Modal
- Edit Profile

The most interesting one is the SignUp page, where I used vuelidate's [async validator](https://monterail.github.io/vuelidate/#sub-asynchronous-validation) feature for checking name availability.

**Ready for review**
_it's branched off #325 New Group Creation, so it will be easier to review after that, but the commits starting from jan 11 are basically about this one_

### CODE REVIEW

**Comment by taoeffect** [comment]
OK, so I had some minor issues with #325, but I decided to merge it because this one (supposedly) builds off of it, and to make your life easier and not screw around with git complexity, I figured the issues that I mentioned in my review comments on #325 can be addressed in this pull request instead.

I posted [6 comments, starting here](https://github.com/okTurtles/group-income-simple/pull/325#pullrequestreview-88651386).

- Please have a look at those and add the changes to this pull request instead of the other one (since that's already merged)
- Also, I'm noticing that for whatever reason, this pull request still shows all of the comments and files as apparently unmerged with respect to `okTurtles:master`. Which is strange since it should show only the changes between this and #325. See if you can figure out why that is (perhaps you have to first merge the changes in `okTurtles:master` into your local branch, and then apply these changes on top of it)

Finally, I didn't get a chance to actually test the new group creation flow, so I'll leave that part for tomorrow and will post any issues I find to this pull request.

Great job so far! 😄 

At least from looking at the code (without actually testing it yet), #325 looked fantastic! 🎉

**Comment by hubudibu** [comment]
So after you merged #325, I could 
- merge upstream master in mine
- rebase this branch on my master
- and push -f it back here so that the confusing history part disappeared 

And now on to the review comments! 🎉

**Comment by hubudibu** [comment]
I addressed all the comments on #325, everything is updated except for [the summary page warnings, see this comment](https://github.com/okTurtles/group-income-simple/pull/325#discussion_r161559956)

I also noticed another bug, that the minimum income field didn't accept decimals, which is not that useful for a mincome set in bitcoin :) now it accepts 2 decimals, based on validation I found on the profile page. let me know if I should increase it.

**Comment by taoeffect** [comment]
> I also noticed another bug, that the minimum income field didn't accept decimals, which is not that useful for a mincome set in bitcoin :) now it accepts 2 decimals, based on validation I found on the profile page. let me know if I should increase it.

Well, in the future I'm guessing the standard BTC unit will be mBTC or something like that, which is easier to deal with. We might want to have "currency" be a class of some sort that includes various info about the currency like:

- Its full name (Bitcoin)
- Its abbreviated name (BTC)
- Its currency symbol (₿ - yes, [there's now a real](https://en.bitcoin.it/wiki/Bitcoin_symbol), dedicated symbol for Bitcoin in Unicode)
- The number of decimal places allowed (e.g. dollars usually allow for two, whereas BTC requires way more, and mBTC can deal with fewer)

File: frontend/simple/views/SignUp.vue [hunk 1]
```diff
@@ -127,47 +166,53 @@ export default {
...
+      name: {
+        required,
+        nonWhitespace,
+        isAvailable (value) {
+          // standalone validator ideally should not assume a field is required
+          if (value === '' || !/^\S+$/.test(value)) return true
+          // async validator can return a promise
+          // we need the opposite of namespace.lookup(value) here
+          return new Promise((resolve, reject) => {
+            namespace.lookup(value).then(reject, resolve)
```
Comment (taoeffect) [inline frontend/simple/views/SignUp.vue:204]: Wow, this is very clean, nicely done.

**Comment by taoeffect** [comment]
So this pull request seems ready to go! Great job! Everything seems to be working correctly! 😄 👍 

The only suggestion I have is changing the title of "Group Advanced Rules" to "Voting Rules". But maybe let's leave that for a future PR? This seems ready to be merged as-is now. 🎉

**Pull request merged**